fix all the lighter and TNT logic being terribly jank.

screret opened the pull request:
## What
removed the same check being copypasted 4 times in LighterBehavior and the creeper mixin doing some really weird stuff

## Implementation Details
did the same thing FlintAndSteelItem (and TntBlock) does

## Outcome
30% less code!!! wow!!! get yours now while stock lasts!!

screret committed:
fix all the lighter and TNT logic being terribly jank.

Spicierspace153 commented:
FLINT.... .and steel

omergunr100 commented on src/main/java/com/gregtechceu/gtceu/common/block/explosive/GTExplosiveBlock.java:
```diff
@@ -105,16 +112,11 @@ public InteractionResult use(BlockState state, Level level, BlockPos pos, Player
-                return List.of();
-            }
+    public void playerWillDestroy(Level level, BlockPos pos, BlockState state, Player player) {
+        if (explodeOnMine && !player.isShiftKeyDown()) {
```
Shouldn't we use the `UNSTABLE` block property instead to match the base game tnt block?

omergunr100 commented on src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/LighterBehavior.java:
```diff
@@ -84,94 +79,48 @@ public InteractionResultHolder<ItemStack> use(Item item, Level level, Player pla
-                return InteractionResult.PASS;
+
+        if ((!canOpen || (tag.getBoolean(LIGHTER_OPEN)) && (player == null || !player.isShiftKeyDown()))) {
+            if ((block instanceof TntBlock || block instanceof GTExplosiveBlock)) {
```
If you decide to use the `UNSTABLE` property this can be a `state.hasProperty(TntBlock.UNSTABLE)` check

screret replied:
well, no, because turns out that most mods' TNT blocks don't implement that either! yay!

omergunr100 commented on src/main/java/com/gregtechceu/gtceu/core/mixins/CreeperMixin.java:
```diff
@@ -1,40 +1,51 @@
+    @Inject(method = "mobInteract",
+            at = @At(value = "FIELD", target = "Lnet/minecraft/world/level/Level;isClientSide:Z", ordinal = 0),
+            cancellable = true)
+    private void gtceu$lighterIgnite(Player player, InteractionHand hand,
```
This probably shouldn't be a mixin, you can add a hook for interacting with the `ItemStack` durability on the component item and call some `IDurabilityBehaviour` for all of these cases.

screret replied:
nope, still need to keep the mixin so empty lighters can't light creepers. but I'll do the other bits.

screret replied:
nvm, found a way.

screret committed:
Merge branch '1.20.1' into sc/fix-lighter-logic

screret committed:
remove charcoal pile igniter checking its own existence when interacted with (???)

screret committed:
clean up other parts of CPI and make the PROGRESS SAVED!? why was it not being saved?!
also made it not extend the workable multi class because it had no reason to do so

screret committed:
make CPI use a tag for the walls (WHY DID IT NOT DO *THAT**!?)

screret committed:
remove mixin for igniting creepers

screret committed:
make lighters able to (hopefully) light any kind of TNT

screret committed:
add block states to GT explosives

screret committed:
Revert "add block states to GT explosives"

This reverts commit c51da38236390c31e3773fdc022393ff96e3228c.

screret committed:
add better igniter tag so that creepers won't delete the lighter you're holding

screret committed:
spotless

screret committed:
copy onion's CPI changes to this PR
